sdk/typespec/typespec_client_core/src/http/sanitizer.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// cspell:ignore traceparent tracestate
+
+use crate::http::{headers::Headers, Url};
+use std::{collections::HashSet, sync::LazyLock};
+
+pub static DEFAULT_ALLOWED_HEADER_NAMES: LazyLock<HashSet<&'static str>> = LazyLock::new(|| {
+    [
+        "accept",
+        "cache-control",
+        "connection",
+        "content-length",
+        "content-type",
+        "date",
+        "etag",
+        "expires",
+        "if-match",
+        "if-modified-since",
+        "if-none-match",
+        "if-unmodified-since",
+        "last-modified",
+        "ms-cv",
+        "pragma",
+        "request-id",
+        "retry-after",
+        "server",
+        "traceparent",
+        "tracestate",
+        "transfer-encoding",
+        "user-agent",
+        "www-authenticate",
+        "x-ms-request-id",
+        "x-ms-client-request-id",
+        "x-ms-return-client-request-id",
+    ]
+    .iter()
+    .copied()
+    .collect()
+});
+
+pub static DEFAULT_ALLOWED_QUERY_PARAMETERS: LazyLock<HashSet<&'static str>> =
+    LazyLock::new(|| ["api-version"].iter().copied().collect());
+
+pub const REDACTED_PATTERN: &str = "REDACTED";
+
+/// A trait that extends a type with sanitization capabilities
+pub trait Sanitizer {
+    /// Sanitizes a type by removing or masking sensitive information based on a set of allowed patterns
+    ///
+    /// # Arguments
+    ///
+    /// * `patterns` - A slice of string slices containing patterns which are allowed in the Url.
+    ///
+    /// # Returns
+    ///
+    /// A String containing the sanitized URL
+    fn sanitize(&self, allowed_patterns: &HashSet<&'static str>) -> String;
```

Comment:
Similarly, maybe just a `&[&'static str]` or even `&[Cow<'static, str>]` might be enough here if we make sure they are sorted going in.

Reply:
Actually, since we're just taking a slice, even `&[&str]` should be fine and would handle this coming from either other idea above anyway, I believe. Keep parameters as simple/accepting as possible.